### PR TITLE
Playbook Utils: New App - PAPP-25471: Initial app release

### PR DESCRIPTION
**Pull request by phantom-bneises**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [X] Testing of all the changes has been performed (for bug fixes / features)
- [X] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [X] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [X] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [x] Verify all checks are passing.
- [X] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [X] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [X] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [x] If this is a new connector or you are adding new actions
    - [X] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [X] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Initial app release

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by ighelani-crest** [comment]
Raised MR to address the static test failures, https://cd.splunkdev.com/phantom/app_release/-/merge_requests/1892

File: playbookutils_connector.py [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+
+        action_map = {
+            'test_connectivity': self._handle_test_connectivity,
+            'get_playbook_tree': self._handle_get_playbook_tree
+        }
+
+        if action_id in action_map:
+            return action_map[action_id](param)
+        else:
+            return self.set_status(phantom.APP_ERROR, 'Action identifier is not available.')
```
Comment (paulnawat) [inline playbookutils_connector.py]: I believe our convention is to return early when possible. What do you think about:

```python
if not action_id in action_map:
  return self.set_status(phantom.APP_ERROR, 'Action identifier is not available.')

return action_map[action_id](param)
```

File: playbookutils.json [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+        },
+        {
+            "action": "get playbook tree",
+            "identifier": "get_playbook_tree",
+            "description": "Get details about the parent/child relationships of playbooks and actions",
+            "type": "generic",
+            "read_only": true,
+            "parameters": {
+                "playbook_run_id": {
+                    "description": "Playbook run ID of a playbook in the tree that is to be retrieved (Default to current PB run if not provided)",
```
Comment (paulnawat) [inline playbookutils.json]: Should we also explain what's considered `current PB` here?
or we should put it in the README.

File: playbookutils_connector.py [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+        except Exception as e:
+            return RetVal(
+                action_result.set_status(phantom.APP_ERROR, f'Error Connecting to server. Details: {e}'),
+                resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _determine_pb_run_id(self, action_result):
+        """ Figure out what the current playbook run id is by using the app_run_id, if it is running in a playbook.
```
Comment (paulnawat) [inline playbookutils_connector.py]: The default PB id logic should be mentioned to the user via README or param description as well.

**Review by paulnawat — CHANGES_REQUESTED** [review]

File: playbookutils_connector.py (hunk 1 above)
Comment (ighelani-crest) [reply playbookutils_connector.py]: Hello @phantom-bneises,
Could you please look into this comment?

File: playbookutils.json (hunk 2 above)
Comment (ighelani-crest) [reply playbookutils.json]: Hello @phantom-bneises,
Could you please look into this comment?

File: playbookutils_connector.py (hunk 3 above)
Comment (ighelani-crest) [reply playbookutils_connector.py]: Hello @phantom-bneises,
Could you please look into this comment?

**Review by paulnawat — APPROVED** [review]
Approved with suggested changes.

File: playbookutils_connector.py (hunk 1 above)
Comment (phantom-bneises) [reply playbookutils_connector.py]: Updated the logic to return early upon error.

File: playbookutils.json (hunk 2 above)
Comment (phantom-bneises) [reply playbookutils.json]: I've extended the action documentation to include additional details in the "verbose" section about this and reworded the parameter description here. Please let me know if that makes sense.

File: playbookutils_connector.py (hunk 3 above)
Comment (phantom-bneises) [reply playbookutils_connector.py]: Extended the method documentation a little further here.

**Review by brian-phantom — APPROVED** [review]

**Comment by ighelani-crest** [comment]
Merging the MR with 2 approvals.

**Pull request merged**